### ISSUE TITLE
Suppress Fidget notifications during InsertMode

### ISSUE THREAD

**Issue by XXiaoA** [j-hui/fidget.nvim#126]
https://user-images.githubusercontent.com/62557596/233656081-5cb777d9-17a2-4183-a860-2d6458c4ab89.mp4

Pylsp not only hints at first time I open a python file, but also hint during typing. It is so annoyed, but I only find the way to disable all pylsp hints by this:
```lua
sources = {
                ["pylsp"] = {
                    ignore = true,
                },
            },
```

Is there a way to solve the problem now?
Version: nightly

**Comment by j-hui** [comment]
Related #122 . #127 might help with this, but unfortunately with the current architecture it's really up to the LSP server to be responsible about not spamming notifications.

**Comment by j-hui** [comment]
Didn't meant to close; I will address this in the rewrite.

**Comment by j-hui** [comment]
Should be possible/much easier with the rewrite, but it won't come out of the box (e.g., you'll need to call `fidget.progress.suppress(true)` on `InsertEnter`). I'll probably add this feature in a future version.

**Comment by j-hui** [comment]
Actually, thinking out loud here (chime in if you have thoughts): what is the right behavior during insert mode?

1. Should we close or freeze the notification window entirely? But there might be useful notifications while in insert mode; some may be unrelated to LSP.
2. Should we drop LSP progress messages while in insert mode? (We could not process them as I suggested earlier, but then we would get a burst of accumulated un-processed messages after leaving insert mode.) But then existing LSP tasks will continue to appear in-progress for the entire duration of being in insert mode.
3. Should we dim or otherwise change the styling of the notification window while in insert mode? To make it less distracting?

Option 2 (modified to only process in-progress tasks) seems to be the right approach, though it's not mutually exclusive with the others. (Option 1 can already be done now using autocommands.)

**Comment by j-hui** [comment]
Closed in 781c8ab221bcd0d803f7fb3efbe114ad94f00c7f (:
